src/grid.cpp

```diff
@@ -547,8 +547,17 @@ diagnostics_t grid::diagnostics(const diagnostics_t &diags) {
 						const real dX[NDIM] = { (x - diags.com[i][XDIM]), (y - diags.com[i][YDIM]), (z - diags.com[i][ZDIM]) };
 						rc.js[i] += dX[0] * U[sy_i][iii] * dV;
 						rc.js[i] -= dX[1] * U[sx_i][iii] * dV;
+						rc.lz2[i] += x * U[sy_i][iii] * dV;
+						rc.lz2[i] -= y * U[sx_i][iii] * dV;
+						rc.lz1[i] += U[lz_i][iii] * dV;
 						rc.gt[i] += dX[0] * G[iiig][gy_i] * dV * rho0;
 						rc.gt[i] -= dX[1] * G[iiig][gx_i] * dV * rho0;
+						const auto ekin = (pow(U[sx_i][iii], 2) + pow(U[sy_i][iii], 2) + pow(U[sz_i][iii], 2)) / 2.0 / U[rho_i][iii] * dV;
+						const auto eint = U[egas_i][iii] * dV - ekin;
```

Comment:
The expression for the internal energy could also be derived from the entropy tracer depending on the value of epsilon. Don't you want to include this in the calculation?